src/App.jsx

```diff
@@ -2,32 +2,66 @@ import React, { useEffect, Fragment, useState } from "react";
 import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 
-function useSemiPersistentState() {
-  const [todoList, setTodoList] = React.useState(() => {
-    const saved = localStorage.getItem("savedTodoList");
-    return saved ? JSON.parse(saved) : [];
-  });
+
+function App() {
+  const [todoList, setTodoList] = React.useState([]) 
+  const [isLoading, setIsLoading] = useState(true); 
 
   useEffect(() => {
-    localStorage.setItem("savedTodoList", JSON.stringify(todoList));
-    console.log("useeffect", todoList);
-  }, [todoList]);
 
-  return [todoList, setTodoList];
-}
+    const myPromise = new Promise((resolve, reject) => {
+      setTimeout(() => {
+      const success = true;               
+      if (success) {
+        resolve({
+          data: {
+            todoList: JSON.parse(localStorage.getItem("list")) || [],
```

Comment:
You are using a different key name to fetch the list item. 

On line 12, you are using 'savedTodoList' as key to save the todo list items, but on line 18, you are using ordinary 'list' as key to fetch the list item. This is cause your app to lose hydration of the todo lists from the localstorage. Kindly correct this.